core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/ToolsResultMemory.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkiverse.langchain4j.runtime.aiservice;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import dev.langchain4j.Experimental;
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.data.message.AiMessage;
+import io.quarkiverse.langchain4j.data.AiStatsMessage;
+
+@Experimental
+public class ToolsResultMemory {
```

Comment:
Again we'll need Javadoc on this as I have zero clue on what the purpose is

Reply:
Yes will add the javadoc when the design will be finalized. But I agree some part could be complex to understant, i will add some docs.

Reply:
TY

Reply:
done